caravel/views.py

```diff
@@ -1238,6 +1237,48 @@ def testconn(self):
                 status=500,
                 mimetype="application/json")
 
+    @api
+    @has_access_api
+    @expose("/warm_up_cache/", methods=['GET', 'POST'])
+    def warm_up_cache(self):
+        """Warms up the cache for the slice or table."""
+        slices = None
+        s = db.session()
+        slice_id = request.args.get('slice_id')
+        table_name = request.args.get('table_name')
+        db_name = request.args.get('db_name')
+
+        if not slice_id and not (table_name and db_name):
+            return json_error_response(__(
+                "Malformed request. slice_id or table_name and db_name "
+                "arguments are expected"), status=400)
+        if slice_id:
+            slices = s.query(models.Slice).filter_by(id=slice_id).all()
+            if not slices:
+                return json_error_response(__(
+                    "Slice %(id)s not found", id=slice_id), status=404)
+        elif table_name and db_name:
+            table = s.query(models.SqlaTable).join(models.Database).filter(
```

Comment:
NIT: the way I've been going about breaking sqla queries is to wrap the statement in parenthesis, and then break before the dot as in javascript when doing method call chains, makes it more readable. Example:
```
foo = (
    session.query(BarModel)
    .filterby(
        id==id,
        bar_style='foobar',
    )
    .all()
```